awswrangler/_distributed.py

```diff
@@ -67,6 +68,9 @@ def set(cls, name: str) -> None:
     @classmethod
     def dispatch_func(cls, source_func: Callable[..., Any], value: Optional[Any] = None) -> Callable[..., Any]:
         """Dispatch a func based on value or the distribution engine and the source function."""
+        if not cls._is_initialized:
```

Comment:
Most of the time we just use `dispatch_on_engine`. Meaning when we call a function with `dispatch_on_engine`, it will figure out which version needs to be called depending on the engine. However, there's a few places where we use `dispatch_func`, such as [here](https://github.com/aws/aws-sdk-pandas/blob/release-3.0.0/awswrangler/distributed/ray/modin/s3/_write_dataset.py#L42). I'm not sure if `dispatch_func` needs to be called there though, or if the function would be dispatched properly anyway. However, without adding this initialization to `dispatch_func`, functions using the code I linked would break.